vllm/engine/protocol.py

```diff
@@ -80,7 +82,10 @@ async def beam_search(
 
         beam_search_params = SamplingParams(logprobs=2 * beam_width,
                                             max_tokens=1,
-                                            temperature=temperature)
+                                            temperature=temperature,
+                                            ignore_eos=ignore_eos,
```

Comment:
here and below, not sure if informing the beam_search_params of `ignore_eos` does anything, since it did not produce the `"stop"` finish reason as I expected.

Reply:
Ok after some more testing this is actually necessary -- otherwise the `eos` ignoring is not happening fully.